From c340807ef03e1ff58da2c10ec9bd1c8ff075fe79 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Tue, 10 Feb 2015 08:27:42 +0800
Subject: [PATCH 1510/1543] MLK-10238-3: ARM: imx: gpc: correct the function
 and structure

commit 065328d5cb7b7455b63dba165e3629a0313596fc from
git://git.freescale.com/imx/linux-2.6-imx.git

Correct the function and structure since update the below patch
(24d70aa Revert "base: power: Add generic OF-based power domain look-up")
(dc092bc PM / Domains: Add generic OF-based PM domain look-up)
Signed-off-by: Robin Gong <b38343@freescale.com>

(cherry picked from commit bddfa2a6ffa844d9a21a63dab974b993e4eccc41)
---
 arch/arm/mach-imx/gpc.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/gpc.c b/arch/arm/mach-imx/gpc.c
index 0f31e86..2d13eb1 100644
--- a/arch/arm/mach-imx/gpc.c
+++ b/arch/arm/mach-imx/gpc.c
@@ -488,7 +488,7 @@ static struct generic_pm_domain *imx_gpc_domains[] = {
 
 static struct genpd_onecell_data imx_gpc_onecell_data = {
 	.domains = imx_gpc_domains,
-	.domain_num = ARRAY_SIZE(imx_gpc_domains),
+	.num_domains = ARRAY_SIZE(imx_gpc_domains),
 };
 
 static int imx_gpc_genpd_init(struct device *dev, struct regulator *pu_reg)
@@ -540,7 +540,7 @@ static int imx_gpc_genpd_init(struct device *dev, struct regulator *pu_reg)
 	pm_genpd_init(&imx6q_pu_domain.base, NULL, is_off);
 	pm_genpd_init(&imx6s_display_domain.base, NULL, is_off);
 
-	return of_genpd_add_provider(dev->of_node, of_genpd_xlate_onecell,
+	return __of_genpd_add_provider(dev->of_node, __of_genpd_xlate_onecell,
 				     &imx_gpc_onecell_data);
 
 }
-- 
1.7.5.4

